server/api/hook.go

```diff
@@ -130,35 +155,38 @@ func PostHook(c *gin.Context) {
 		return
 	}
 
-	if tmpPipeline == nil {
+	if pipelineFromForge == nil {
 		msg := "ignoring hook: hook parsing resulted in empty pipeline"
 		log.Debug().Msg(msg)
 		c.String(http.StatusOK, msg)
 		return
 	}
-	if tmpRepo == nil {
+	if repoFromForge == nil {
 		msg := "failure to ascertain repo from hook"
 		log.Debug().Msg(msg)
 		c.String(http.StatusBadRequest, msg)
 		return
 	}
 
 	//
-	// 2. Get related repo from store and take repo renaming into account
+	// 3. Check the repo from the token is matching the repo returned by the forge
 	//
 
-	repo, err := _store.GetRepoNameFallback(tmpRepo.ForgeRemoteID, tmpRepo.FullName)
-	if err != nil {
-		log.Error().Err(err).Msgf("failure to get repo %s from store", tmpRepo.FullName)
-		handleDBError(c, err)
+	if repo.ForgeRemoteID != repoFromForge.ForgeRemoteID {
+		log.Debug().Msgf("ignoring hook: repo %s does not match the repo from the token", repo.FullName)
```

Comment:
I would probably raise this to warning as this is indication of either misconfiguration or something else bad that should be looked into

Reply:
```suggestion
		log.Warn().Msgf("ignoring hook: repo %s does not match the repo from the token", repo.FullName)
```